service/controller/v18/cloudconfig/spec.go

```diff
@@ -42,6 +47,37 @@ func (e *baseExtension) templateData() TemplateData {
 	return data
 }
 
+func (e *baseExtension) encryptAndGzip(ctx context.Context, data []byte) ([]byte, error) {
+	var encrypted []byte
+	{
+		e, err := e.encrypter.Encrypt(ctx, e.encryptionKey, string(data))
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		encrypted = []byte(e)
+	}
+
+	var gzipped []byte
+	{
+		buf := new(bytes.Buffer)
+		w := gzip.NewWriter(buf)
+
+		_, err := w.Write(encrypted)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		err = w.Close()
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		gzipped = buf.Bytes()
+	}
+
+	return gzipped, nil
+}
```

Comment:
Encryption is done within this package. No weird TLS indirection in `encrypter` package.